pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -89,17 +90,26 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	}
 
 	// manage status
-	if actualDaemonSet.Status.NumberAvailable > 0 {
+	var availabilityErrors []string
+	if actualDaemonSet.Status.NumberAvailable == 0 {
+		availabilityErrors = append(availabilityErrors, fmt.Sprintf("daemonset/%v: no daemon pods available on any node.", actualDaemonSet.Name))
+	}
+	for _, currResult := range directResourceResults {
```

Comment:
you need to make sure all of these are Conditions["Available"] == true https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workload_controller_openshiftapiserver_v311_00.go#L214-L227